src/main/java/org/apache/sling/resourceresolver/impl/mapping/Interpolator.java

```diff
@@ -0,0 +1,154 @@
+/*
```

Comment:
It's ok to have a copy here for getting the functionality in, but we should rather directly use the class from Apache Felix by embedding just that class (and using the maven shade plugin to move it to a different package). But we can do this in a follow up commit

Reply:
The class is not exported from the Bundle (AFAI can tell) and so I could not use it.

Reply:
I reverted everything back to the code from Felix with the exception of added logging statements and constants

Reply:
We can embed it, exactly because it is not exported - but there is no strict need for it, other than to avoid code duplication - and as mentioned we can do this later on